src/main/java/Duke.java

```diff
@@ -1,10 +1,162 @@
+     *
+     * @param taskDescription TaskDescription Stores the description of the task.
+     * @param taskDueBy       TaskDueBy stores the deadline of the task which is created as a deadline.
+     * @param taskDueAt       TaskDueAt stores the event timing of the task created as an event.
+     * @param isTaskValid     IsTaskValid stores true if the task statement emtered by the user is a valid task creation statement and fasle, otherwise.
+     * @param index           Index stores the index of the "/" in the entered String
+     */
+    private static void addTaskToList(String userInput) {
+
+        String taskDescription;
+        String taskDueBy;
+        String taskDueAt;
+        int index;
+        boolean isTaskValid = true;
+
+        if (userInput.startsWith("todo")) {
+            scheduledTasks[taskCounter] = new Todo(userInput);
+        } else if (userInput.startsWith("deadline")) {
+            index = userInput.indexOf("/");
+            scheduledTasks[taskCounter] = new Deadline(userInput.substring(0, index), userInput.substring(index + 3));
+        } else if (userInput.startsWith("event")) {
+            index = userInput.indexOf("/");
+            scheduledTasks[taskCounter] = new Event(userInput.substring(0, index), userInput.substring(index + 3));
+        } else {
+            isTaskValid = false;
+            System.out.println("Invalid task creation statement");
+        }
+
+        if (isTaskValid) {
+            taskCounter++;
+            printLine();
+            System.out.println("Got it. I've added this task:");
+            System.out.println(" " + scheduledTasks[taskCounter - 1]);
+            System.out.println("Now you have " + taskCounter + " tasks in the list.");
+        }
+    }
+
+    /**
+     * Updates the status of the task by marking it as done in the task list.
+     *
+     * @param taskNumberCompleted TaskNumberCompleted stores the task number which has been completed by the user.
+     */
+    private static void markTaskAsDone(String userInput) {
+        printLine();
+        int taskNumberCompleted = Integer.parseInt(userInput.substring(userInput.indexOf(" ") + 1));
+
+        if ((taskNumberCompleted <= taskCounter) && (taskNumberCompleted > 0)) {
+            scheduledTasks[taskNumberCompleted - 1].markAsDone();
+            System.out.println("Nice! I have marked this task as done:");
+            System.out.println(scheduledTasks[taskNumberCompleted - 1]);
+        } else {
+            System.out.println("Sorry, no task is assigned at this number, you might want to re-check?");
+        }
+    }
+
+    /**
+     * Lists all the tasks in the task list along with their task completion status.
+     * The tasks are enlisted which reveal if they are a "todo", "event" or a "deadline".
+     *
+     * @param taskCompletionStatus TaskCompletionStatus stores true if the task is completed, false otherwise.
+     */
+    private static void List() {
+        int i;
```

Comment:
As the integer `i` is only used in the for loop in this function, this variable should be defined in the least possible scope (only define `i` in the for loop)

Reply:
Okay sure.